internal/services/hdinsight/hdinsight_cluster_pool_resource.go

```diff
@@ -0,0 +1,464 @@
+}
+
+type ClusterPoolProfile struct {
+	ClusterPoolVersion string `tfschema:"cluster_pool_version"`
+}
+
+type ComputeProfile struct {
+	VmSize string `tfschema:"vm_size"`
+}
+
+type LogAnalyticsProfile struct {
+	LogAnalyticsProfileEnabled bool   `tfschema:"log_analytics_profile_enabled"`
+	WorkspaceId                string `tfschema:"workspace_id"`
+}
+
+type NetworkProfile struct {
+	PrivateApiServerEnabled bool   `tfschema:"private_api_server_enabled"`
+	OutboundType            string `tfschema:"outbound_type"`
+	SubnetId                string `tfschema:"subnet_id"`
+}
+
+var _ sdk.ResourceWithUpdate = ClusterPoolResource{}
+
+func (r ClusterPoolResource) ModelObject() interface{} {
+	return &ClusterPoolModel{}
+}
+
+func (r ClusterPoolResource) ResourceType() string {
+	return "azurerm_hdinsight_cluster_pool"
+}
+
+func (r ClusterPoolResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return hdinsights.ValidateClusterPoolID
+}
+
+func (r ClusterPoolResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"compute_profile": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"vm_size": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"managed_resource_group_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"cluster_pool_profile": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"cluster_pool_version": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"log_analytics_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"log_analytics_profile_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"workspace_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"network_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
```

Comment:
and here as well, shouldn't this have a `MaxItems` property defined for the list?